docs/area-chart-stacked/README.md

```diff
@@ -4,11 +4,45 @@
 
 ### Data structure
 
-This component requires an array of layer objects, where each layer object represents a layer in the stack.
+The stackedArea component requires an stacked structure of data. The structure converts a flat data array into an array of layer arrays, where each layer array represents a category or component of the total quantity. Each layer array should have a `key` property, which is used to identify the layer in the data join. The layer array is then made up of computed values for the _y0_ and _y1_ positions of the area, which are used to draw the area chart as well as a `data` property, which is the original data object that the layer was computed from.
+
+```code
+const stackedData = [
+  [
+      [0, 10, data: {...}],
+      [0, 20, data: {...}]
+      key: "key1"
+  ],
```

Comment:
At first I was like… hey, this is not valid JavaScript. But then I noticed that it's difficult to express the data as a JS expression (while keeping it readable) :(

Reply:
its even weirder when you want to express it in typescript:

```typescript
interface StackedData<T extends Record> extends Array<number> {
  0: number;
  1: number;
  data: T;
  key: string;
}
```
<img width="599" alt="Screenshot 2024-10-30 at 22 08 05" src="https://github.com/user-attachments/assets/6b7eeb77-a0b5-4398-b2ac-b4c1a7016f4e">